sql/src/main/java/io/crate/analyze/AnalyzedCreateAnalyzer.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.analyze;
+
+import io.crate.expression.symbol.Symbol;
+import io.crate.sql.tree.GenericProperties;
+import io.crate.sql.tree.GenericProperty;
+import org.elasticsearch.common.collect.Tuple;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AnalyzedCreateAnalyzer implements DDLStatement {
+
+    private final String ident;
+    @Nullable
+    private final String extendedAnalyzerName;
+    @Nullable
+    private Tuple<String, GenericProperties<Symbol>> tokenizer;
+    private final GenericProperties<Symbol> genericAnalyzerProperties;
+    private final Map<String, GenericProperties<Symbol>> charFilters;
+    private final Map<String, GenericProperties<Symbol>> tokenFilters;
+
+    AnalyzedCreateAnalyzer(String ident, @Nullable String extendedAnalyzerName) {
+        this.ident = ident;
+        this.extendedAnalyzerName = extendedAnalyzerName;
+        this.genericAnalyzerProperties = new GenericProperties<>();
+        this.charFilters = new HashMap<>();
+        this.tokenFilters = new HashMap<>();
+    }
+
+    void tokenDefinition(String name, GenericProperties<Symbol> properties) {
+        tokenizer = new Tuple<>(name, properties);
+    }
+
+    void addTokenFilter(String name, GenericProperties<Symbol> properties) {
+        tokenFilters.put(name, properties);
+    }
+
+    void addGenericAnalyzerProperty(GenericProperty<Symbol> property) {
+        this.genericAnalyzerProperties.add(property);
+    }
+
+    void addCharFilter(String name, GenericProperties<Symbol> properties) {
```

Comment:
Could we make the analyzed statement class itself immutable ?

With prepared statements we can cache the analyzed statements and re-use them across query execution. Having this immutable would eliminate one whole class of bugs.

Reply:
yep, will do